app/src/main/java/com/nextcloud/talk/controllers/CallController.java

```diff
@@ -433,7 +433,14 @@ private void basicInitialization() {
     }
 
     private void handleFromNotification() {
-        ncApi.getRooms(credentials, ApiUtils.getUrlForGetRooms(baseUrl))
+        Integer apiVersion = ApiUtils.getApiVersion(conversationUser, "conversation", new int[] {1});
+
+        if (apiVersion == null) {
+            Log.e(TAG, "No supported API version found", new Exception("No supported API version found"));
```

Comment:
There is no reason to add an exception the message is enough since this only leads to printing the exception that offers no info that the original message didn't :)

Reply:
Apart from the stack trace which helps you too find where the problem comes from?

Reply:
Since it's also kind of fatal and the code will not continue we could also throw?

Reply:
Of course it will continue, you didn't call throw exception!

Reply:
but we return :)

Reply:
Okay, so I add a throw everywhere. Might be saver anyway

Reply:
> Okay, so I add a throw everywhere. Might be saver anyway

If you throw it then don't log it ;) - A "typical" Java pattern would be "log or throw" as in if you "handle the exception" then fell free to log it (where you handle it)

> but we return :)

That is hacky ;)

> Okay, so I add a throw everywhere. Might be saver anyway

Yeah, that is the nicer way to handle it while "Exception" is very generic and will create warnings, so we might consider creating our own (or find a suiting one)